client/src/components/annotators/annotator.js

```diff
@@ -0,0 +1,95 @@
+import geo from "geojs";
+import Vue from "vue";
+
+import { throttle } from "lodash";
+
+export default {
+  props: {
+    frameRate: {
+      type: Number,
+      required: true
+    }
+  },
+  provide() {
+    return {
+      annotator: this.provided
+    };
+  },
+  data() {
+    this.provided = new Vue({
+      computed: {
+        viewer: () => this.viewer,
+        playing: () => this.playing,
+        frame: () => this.frame,
+        maxFrame: () => this.maxFrame,
+        syncedFrame: () => this.syncedFrame
+      }
+    });
+    return {
+      ready: false,
+      playing: false,
+      frame: 0,
+      maxFrame: 0,
+      syncedFrame: 0
+    };
+  },
+  created() {
+    this.provided.$on("prev-frame", this.prevFrame);
+    this.provided.$on("next-frame", this.nextFrame);
+    this.provided.$on("play", this.play);
+    this.provided.$on("pause", this.pause);
+    this.provided.$on("seek", this.seek);
+    this.emitFrame();
+    this.emitFrame = throttle(this.emitFrame, 200);
+  },
+  methods: {
+    baseInit() {
```

Comment:
Might be good to add a comment to this function as well explaining what it's doing, if possible.

Reply:
this is explained in `src/components/annotators/README.md`